### ISSUE TITLE
Restream to Twitch with a separate VOD audio channel

### ISSUE THREAD

**Issue by erickskrauch** [ossrs/srs#2288]
_I'm sorry that I had to broke your issue template, it was in Chinese._

I'm trying to achieve a somewhat strange idea: I want to stream on Twitch and YouTube, but in such a way that on Twitch the broadcast will use [their feature of a separate track for VOD](https://help.twitch.tv/s/article/soundtrack-audio-configuration#obs) ( to exclude music) and on YouTube with hat particular track without music.

I have already tricked OBS by specifying the server I was not using as the platform to broadcast to:
![OBS Twitch broadcast setup](https://user-images.githubusercontent.com/4787256/115098307-ac6c7380-9f2f-11eb-88b3-72b244c35da8.png)
![OBS output settings](https://user-images.githubusercontent.com/4787256/115098314-baba8f80-9f2f-11eb-9512-567c8bc122a9.png)

Then in the `hosts` file I override the address for this server, directing it to `localhost`:
![hosts file configuration](https://user-images.githubusercontent.com/4787256/115098347-edfd1e80-9f2f-11eb-8790-ad40c90f0001.png)

Now I'm ready to receive the incoming stream and process it in my way. But the problem is that I can't catch the second track in any way. No matter which side I go to, I always get only 1 track. The second track is nowhere to be found. I suspect that for some reason the SRS implementation itself cuts off everything but the first track.

I checked: if I'll remove the redirection to `localhost` and start streaming directly on Twitch, the separate VOD track works fine. OBS has no way of knowing which end server it's streaming to, so the `rtmp` server gets the same content and the problem is somewhere on its side.

Is there anything I can do about it?

**Comment by winlinvip** [comment]
I didn't get you, what's the deploy arch? Use OBS to publish to SRS or to Twitch? Or publish to SRS to forward to twitch? What's your use scenario? Please don't only tell us a feature to implements.

**Comment by erickskrauch** [comment]
Sorry to confuse you. I'm trying to stream into multiple platforms, so I need to stream to the SRS server, which then will publish a stream to the target platforms. The trick with "broadcast to twitch" from the OBS is only necessary to activate the "Twitch VOD Track" feature, but actually, the stream is going to the SRS server.

**Comment by winlinvip** [comment]
OK, I got you. You publish stream to SRS by OBS, then use SRS to forward to multiple platforms.

Is it possible to publish to Twitch by normal RTMP stream? Like you select the `Custom RTMP server` of OBS?

**Comment by erickskrauch** [comment]
Yes, it is possible, but then I can't specify a separate audio track for VOD in the video settings tab. This is a nuance of the OBS implementation: it only shows this option if it thinks the stream is going to be published on Twitch. That's why I need this hack with redefining the server address through the hosts file.

In any case, this hack works correctly and the stream comes to the SRS server. The problem is that there is no way I can get to the second track on the SRS side.

**Comment by erickskrauch** [comment]
@winlinvip, sorry for pinging. Do you have any ideas about this?

**Comment by winlinvip** [comment]
Yes, I think we should make it better to forward to streaming platform, like twitch etc.

I mark this issue to feature, and we may check it in SRS 5.0, thanks.

**Comment by krakow10** [comment]
I am interested in the same use case.
OBS -> SRS (audio track 1 with music, audio track 2 with no music)
SRS -> Youtube (no music)
SRS -> Twitch (with music for live, no music for VOD track)

I think streaming to SRS using SRT might make more sense than trying to accept the specialized Twitch VOD RTMP from OBS which is limited to 2 audio tracks.  This way, only the specialized Twitch VOD RTMP _output_ with 2 audio tracks has to be implemented.

OBS has a [pull request](https://github.com/obsproject/obs-studio/pull/9028) to be able to stream SRT with multiple audio tracks, with [build artifacts](https://github.com/obsproject/obs-studio/actions/runs/5191672359?pr=9028) also available.

This is also possible by using the OBS ffmpeg custom output but it is not nearly as stable.
